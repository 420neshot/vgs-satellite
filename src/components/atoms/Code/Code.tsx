import React from 'react';
import classnames from 'classnames';
import { Button } from 'reactstrap';
import Icon from 'src/components/atoms/Icon/Icon';
import { notify } from 'src/redux/utils/notifications';
import { CopyToClipboard } from 'react-copy-to-clipboard';

import Prism from 'prismjs/prism';
import 'prismjs/components/prism-yaml';
import 'prismjs/components/prism-bash';
import 'prismjs/components/prism-python';
import 'prismjs/components/prism-java';
import 'prismjs/components/prism-ruby';
import 'prismjs/components/prism-markup';
import 'prismjs/components/prism-markdown';
import 'prismjs/components/prism-json';

export interface ICodeProps {
  numbers?: boolean;
  children?: string;
  language?: string;
  inline?: boolean;
  hideCopy?: boolean;
  onCopy?: () => void;
  icon?: string;
  className?: string;
}

export default class Code extends React.Component<ICodeProps> {
  static defaultProps = {
    numbers: true,
    language: 'markup',
    inline: false,
  };

  componentDidMount() {
    Prism.highlightAll();
  }

  componentDidUpdate() {
    Prism.highlightAll();
  }

  render() {
    if (!this.props.inline) {
      return (
        <pre
          className={classnames(
            { 'line-numbers': this.props.numbers },
            this.props.className,
          )}
        >
          {this.props.icon &&
            <Icon name={this.props.icon} type="text" position="left"/>
          }
          <code className={`language-${this.props.language}`}>
              {this.props.children}
          </code>
          {!this.props.hideCopy &&
            <CopyToClipboard
                text={this.props.children}
                onCopy={() => {
                  notify.success('Copied to clipboard');
                  if (this.props.onCopy) {
                    this.props.onCopy();
                  }
                }}
            >
              <Button color="link" size="sm" className={'position-absolute top right mt-2 text-muted'}>
                  <Icon name="copy" type="text" position="left" />Copy
              </Button>
            </CopyToClipboard>
          }
        </pre>
      );
    } else {
      return (
        <code
          className={classnames(
            `language-${this.props.language}`,
            this.props.className,
          )}
        >
          {this.props.children}
        </code>
      );
    }
  }
}
