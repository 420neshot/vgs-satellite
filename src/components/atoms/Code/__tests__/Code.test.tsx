import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Code from '../Code';
const renderer = new ShallowRenderer();

describe('Code renders', () => {
  const sampleYaml = 'data:\n- attributes:\n';

  it('should render correctly with numbers', () => {
    const component = renderer.render(
      <Code numbers language="yaml">{sampleYaml}</Code>,
    );
    expect(component).toMatchSnapshot();
  });
});
