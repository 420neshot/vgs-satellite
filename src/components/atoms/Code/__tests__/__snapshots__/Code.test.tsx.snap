// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code renders should render correctly with numbers 1`] = `
<pre
  className="line-numbers"
>
  <code
    className="language-yaml"
  >
    data:
- attributes:

  </code>
  <CopyToClipboard
    onCopy={[Function]}
    text="data:
- attributes:
"
  >
    <Button
      className="position-absolute top right mt-2 text-muted"
      color="link"
      size="sm"
      tag="button"
    >
      <Icon
        id=""
        name="copy"
        position="left"
        type="text"
      />
      Copy
    </Button>
  </CopyToClipboard>
</pre>
`;
